slides/bitcoin_network.tex

```diff
@@ -0,0 +1,171 @@
+	\titlepage
+\end{frame}
+
+%%%
+\begin{frame}{Bitcoin is a Peer-to-Peer Network}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/p2p}
+		\color{focus}
+		\node (text) at (-0.5,3.5) { Nodes };
+		\draw[->, thick]	(text.south) -- (agenta.west);
+		\draw[->, thick]	(text.south) -- (agentb.north west);
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Joining the Network}
+	\begin{center}
+		\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+			\input{../assets/figures/join_network}
+		\end{tikzpicture}
+	\end{center}
+
+	\vspace{1cm}
+	
+	After the initial handshake, we request a list of addresses and connect to more peers.
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Functionality of a Full Node}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/node_functions}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+
+%%%
+\begin{frame}{Verify New Blocks}
+	
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/full_node_functions}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Distribute Verified Blocks}
+	
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/full_node_functions2}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Wallet Functionality}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/wallet_functions}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+\begin{frame}{Mining Functionality}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/mining_functions}
+	\end{tikzpicture}
+\end{frame}
+
+%%%
+\begin{frame}{Node Distribution}
+	\begin{figure}[t]
+		\includegraphics[width = 0.9\textwidth]{../assets/images/node_distribution}
+		Source: \link \url{https://bitnodes.io/}, March 2021
+	\end{figure}
+
+	\begin{footnotesize}
+		\begin{itemize}
+			\item $\sim$50\% of nodes in Europe
+			\item $\sim$20\% of nodes in the USA
+		\end{itemize}
+	\end{footnotesize}
+	
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Historic Number of Full Nodes}
+	\begin{description}
+		\item[\textbf{Dec 2013:}] $\sim$15500 full nodes
+		\item[\textbf{Dec 2015:}] $\sim$5500 full nodes
+		\item[\textbf{Mar 2017:}] $\sim$6500 full nodes
+		\item[\textbf{Mar 2019:}] 10362 full nodes
+		\item[\textbf{Mar 2021:}] 10179 full nodes
+	\end{description}	
+		
+	\vspace{1.5 cm}
+		
+	Reasons for decline?
+	
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Network Partitions}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/partitioned_network}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+
+%%%
+\begin{frame}{Extended Network}
+	\centering
+	\begin{tikzpicture}[scale=1, every node/.style={scale=1}]
+		\input{../assets/figures/extended_network}
+	\end{tikzpicture}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Simplified Payment Verification}
+
```

Comment:
Less Centralized options:
* Simplified Payment Verification (SPV).
* Running full node at home and connect to it with mobile devices.